src/rust/engine/src/selectors.rs

```diff
@@ -59,6 +59,9 @@ impl Selector {
     )
   }
 
+  /*
+  * The product type this selector will ultimately produce.
```

Comment:
I know we're not really following rust style at all, but the indentation is funny here.